src/VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleProductsController.cs

```diff
@@ -282,6 +282,36 @@ public async Task<ActionResult> DeleteProduct([FromQuery] string[] ids)
             return Ok();
         }
 
+        /// <summary>
+        /// Plenty delete the specified items by id.
+        /// </summary>
+        /// <param name="ids">The items ids.</param>
+        [HttpPost]
+        [Route("plentyDelete")]
+        public async Task<ActionResult> PlentyDeleteProducts([FromBody] string[] ids)
+        {
+            const int batchSize = 20;
+            var products = new List<CatalogProduct>();
+
+            for (var i = 0; i < ids.Length; i += batchSize)
+            {
+                var idsToDelete = ids.Skip(i).Take(batchSize);
+                products.AddRange(await _itemsService.GetByIdsAsync(idsToDelete.ToArray(), ItemResponseGroup.ItemInfo.ToString()));
+                var authorizationResult = await _authorizationService.AuthorizeAsync(User, products, new CatalogAuthorizationRequirement(ModuleConstants.Security.Permissions.Delete));
+                if (!authorizationResult.Succeeded)
+                {
+                    return Unauthorized();
+                }
+            }
+
+            for (var i = 0; i < ids.Length; i += batchSize)
```

Comment:
Why do we need the second cycle? Couldn't it be made in one?

Reply:
If user haven't permission for one item from the array we throw exception. So we can delete all products only when permission for every product successfully checked.

Reply:
Indeed.
I think you should not use `products` here, use `ids` instead in the second cycle.
Products array is not supposed to be changed.

Reply:
changed